internals/cli/error_handler.go

```diff
@@ -12,7 +10,7 @@ func (c *CommandClause) argumentError(args []string) error {
 	}
 	errorText, minimum, maximum := "", getRequired(c.Args), len(c.Args)
 
-	if strings.Contains(reflect.TypeOf(c.Args[0].Value).String(), "List") {
+	if c.Args[0].Value.String() == new(StringListValue).String() {
```

Comment:
If we have a first argument that has `StringListValue` as a default value, the user would always get the error below. Of course, this is very unlikely to happen, but the fact that it is possible, points at the fact that this comparison is a little out of place.

If we zoom out a little, we can see that the information whether the function is run on a command that has arguments that take 1 or more values is available to the caller of this function. So if we just pass that information to this function, we do not need this comparison at all.

Since this error is _only_ relevant to arguments that take multiple values, we can even split the validation in `argumentError()` to multiple functions. I've made a suggestion for this in 
2b62d06 (and 34eac77 also contains 2 fixes because I am a retard and have trouble committing my code correctly :grin: ). Have a look and see what you think of this approach.

Reply:
Indeed, the approach is way nicer that complicating to adjust one function to do it all.
I haven't thought of separating the functionality in the first place.
Will solve this discussion and thanks for the help ^-^.